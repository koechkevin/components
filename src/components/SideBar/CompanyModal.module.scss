@import '../../styles/_global.scss';

$badgeHeight: 18px;

.mask {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 1111;
  background-color: rgba($color: $color-black, $alpha: 0.6);
}

.companyModal {
  position: absolute;
  top: 48px;
  left: 50%;
  width: 312px;
  border-radius: 12px;
  transform: translateX(-50%);
  background-color: $color-base-dark;
}

.listItem {
  padding: 16px;

  &:not(:last-child) {
    border-bottom: 1px solid $color-base-darker;
  }

  .avatar {
    border-radius: 6px;
    border: 3px solid $color-base-dark;
    background-color: $color-dark-gray;
    transition: all 0.4s cubic-bezier(0.075, 0.82, 0.165, 1);

    > img {
      border-radius: 6px;
      border: 2px solid $color-base-dark;
    }

    :global {
      .ant-avatar-string {
        line-height: 34px !important;
      }
    }
  }

  &.active .avatar {
    border-width: 2px;
    border-color: $color-white;
  }

  .info {
    flex: 1 auto;
    margin-left: 16px;
    margin-right: 16px;
    overflow: hidden;
    font-size: 16px;

    :global(.ant-typography) {
      color: $color-white;
    }
  }
}

.badge {
  padding: 2px 5px;
  line-height: 1;
  font-size: 12px;
  height: $badgeHeight;
  color: $color-white;
  border-radius: 2px;
  background-color: $color-buttercup;

  @include display-flex();
  @include flex-center();
}
