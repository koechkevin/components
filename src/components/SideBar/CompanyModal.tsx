import { Col, Row } from 'antd';
import React, { FC, useEffect, useState } from 'react';
import ReactDOM from 'react-dom';

import { getNameInitials } from '../../utils';
import { Avatar } from '../Avatar';
import { Text } from '../Typography';
import styles from './CompanyModal.module.scss';

export interface ListItem {
  name: string;
  count?: number;
  companyId?: string;
  avatarUrl: string;
  avatarColor?: string;
}

interface Props {
  list?: any[];
  visible: boolean;
  companyId?: string;
  onClose?: (visible?: boolean) => void;
  onSelect?: (item: ListItem) => void;
}

export const CompanyModal: FC<Props> = (props) => {
  const { list, visible, companyId, onClose, onSelect } = props;
  const [display, setDisplay] = useState(false);

  useEffect(() => {
    setDisplay(visible);
  }, [visible]);

  const handleSelect = (item: ListItem) => {
    onSelect && onSelect(item);
  };

  const handleClose = () => {
    setDisplay(false);
    onClose && onClose(false);
  };

  const modal = display && (
    <div className={styles.mask} onClick={handleClose}>
      <Row className={styles.companyModal}>
        {list && list.length
          ? list.map((item: ListItem, index: number) => (
              <Row
                key={index}
                type="flex"
                align="middle"
                onClick={() => handleSelect(item)}
                className={[styles.listItem, companyId === item.companyId ? styles.active : ''].join(' ')}
              >
                <Avatar
                  size={38}
                  shape="square"
                  src={item.avatarUrl}
                  className={styles.avatar}
                  color={item.avatarColor}
                >
                  {getNameInitials(item.name)}
                </Avatar>
                <Col className={styles.info}>
                  <Text>{item.name}</Text>
                </Col>
                {item.count && <span className={styles.badge}>{item.count < 100 ? item.count : '+99'}</span>}
              </Row>
            ))
          : null}
      </Row>
    </div>
  );

  return ReactDOM.createPortal(modal, document.body);
};

export default CompanyModal;
