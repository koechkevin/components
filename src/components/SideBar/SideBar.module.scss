@import '../../styles/_global.scss';

.sideBar {
  height: 100%;
  overflow: auto;
  color: $color-white;

  &:global {
    &.ant-layout-sider-dark {
      background-color: rgba($color: $color-side-bar-dark, $alpha: 1);

      >.ant-layout-sider-children {
        max-width: max-content;
        min-width: 100%;
      }
    }
  }
}

.sideHeader {
  position: relative;
  padding: 16px;
  min-width: 200px;
  margin-bottom: 16px;
  background-color: $color-base-dark;

  @include respond-to('xs-l') {
    height: 72px;
  }

  &:global(.ant-row-flex) {
    flex-wrap: nowrap;
    overflow: hidden;
  }

  .profile {
    @include display-flex();
    @include align-items(center);
    flex: 1 auto;
    overflow: hidden;

    :global(.ant-col) {
      flex: 1 auto;
      @include display-flex();
      @include align-items(center);
      overflow: hidden;

      b {
        display: inline-block;
        max-width: 120px;
        @include ellipsis();
      }
    }
  }

  .menuBtn {
    color: $color-base-gray;
  }

  .closeBtn {
    background-color: $color-base-darker;
    color: $color-grey-lighter;
  }

  .caretDown {
    color: $color-grey-lighter;
  }

  .name {
    font-size: 16px;
  }
}