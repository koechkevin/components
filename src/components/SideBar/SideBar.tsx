import { Layout } from 'antd';
import { CollapseType } from 'antd/lib/layout/Sider';
import React, { CSSProperties, FC } from 'react';

import styles from './SideBar.module.scss';

const { Sider } = Layout;

interface Props {
  style?: CSSProperties;
  className?: string;
  width?: number | 200;
  collapsed?: boolean;
  theme?: 'light' | 'dark';
  onCollapse?: (collapsed: boolean, type: CollapseType) => void;
}

export const SideBar: FC<Props> = (props) => {
  const { theme, width, onCollapse, collapsed, children, className, ...restProps } = props;
  const cls = [styles.sideBar, className ? className : ''].join(' ');

  return (
    <Sider
      collapsible
      theme={theme}
      width={width}
      trigger={null}
      className={cls}
      collapsedWidth={0}
      collapsed={collapsed}
      onCollapse={onCollapse}
      {...restProps}
    >
      {children}
    </Sider>
  );
};

export default SideBar;
