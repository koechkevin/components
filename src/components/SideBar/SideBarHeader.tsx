import { faStream, faTimes } from '@fortawesome/pro-regular-svg-icons';
import { faCaretDown } from '@fortawesome/pro-solid-svg-icons';
import { Col, Row } from 'antd';
import React, { FC } from 'react';

import { getNameInitials, ProductType } from '../../utils';
import { Avatar } from '../Avatar';
import { Icon } from '../Icon';
import styles from './SideBar.module.scss';

export interface Props {
  name?: string;
  logo?: string;
  isMobile?: boolean;
  avatarColor?: string;
  productId?: ProductType;
  onOpen?: () => void;
  onCollapse?: () => void;
}

export const SideBarHeader: FC<Props> = (props) => {
  const { name, avatarColor, productId, isMobile, logo, onOpen, onCollapse } = props;
  const isCandidate = productId === ProductType.Candidate;

  return (
    <Row className={styles.sideHeader} type="flex" align="middle" justify="space-between">
      <Col className={styles.profile}>
        {!isCandidate || isMobile ? (
          <Avatar size={isCandidate ? 'medium' : 'small'} src={logo} color={avatarColor} shape="square">
            {getNameInitials(name)}
          </Avatar>
        ) : null}
        <Col onClick={onOpen}>
          {name && (
            <b className={styles.name} style={{ marginLeft: !isCandidate || isMobile ? 8 : 0, marginRight: 16 }}>
              {name}
            </b>
          )}
          {isMobile && isCandidate && <Icon icon={faCaretDown} className={styles.caretDown} />}
        </Col>
      </Col>
      <Col>
        <Icon
          onClick={onCollapse}
          style={{ marginRight: -8 }}
          icon={isMobile ? faTimes : faStream}
          className={isMobile ? styles.closeBtn : styles.menuBtn}
        />
      </Col>
    </Row>
  );
};

export default SideBarHeader;
