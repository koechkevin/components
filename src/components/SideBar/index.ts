export { default as SideBar } from './SideBar';
export { default as CompanyModal } from './CompanyModal';
export { default as SideBarHeader } from './SideBarHeader';