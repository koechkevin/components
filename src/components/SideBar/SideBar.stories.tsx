import { faHome, faSuitcase } from '@fortawesome/pro-regular-svg-icons';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { action } from '@storybook/addon-actions';
import { boolean, object } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import logo from '../../icons/logo.svg';
import mockLogo from '../../icons/mocks/company.png';
import { getFlatMenuKeys, ProductType } from '../../utils';
import { BaseMenu, ChannelMenu } from '../Menu';
import { ChannelItem, MenuItemProps } from '../Menu/interface';
import CompanyModal, { ListItem } from './CompanyModal';
import SideBar from './SideBar';
import SideBarHeader from './SideBarHeader';

const collapsed = boolean('collapsed', false);
const companyList: ListItem[] = [
  { companyId: '123', avatarUrl: logo, name: 'Airbus', count: 4, avatarColor: '#fb6d77' },
  { companyId: '1234', avatarUrl: mockLogo, name: 'Pepsi', count: 21, avatarColor: '#46a094' },
  { companyId: '12345', avatarUrl: '', name: 'Airbus', avatarColor: '#5d85ce' },
];

const menu = [
  {
    name: 'Home',
    icon: faHome,
    path: '/app/home',
  },
  {
    name: 'Jobs',
    icon: faSuitcase,
    path: '/app/jobs',
  },
];
const channels: ChannelItem[] = [
  {
    id: '1',
    name: 'Direct Message 1',
    path: '/1',
    isTyping: true,
  },
  {
    id: '2',
    name: 'Example Message 2',
    path: '/2',
  },
  {
    id: '3',
    name: 'Example Message 3',
    path: '/3',
  },
];
const jobsMenu: MenuItemProps[] = [
  {
    name: 'Base Menu',
    path: '/app/example',
    routes: [
      {
        name: 'Page 1',
        path: '/app/example/page-1',
      },
      {
        name: 'Page 2',
        path: '/app/example/page-2',
      },
      {
        name: 'Page 3',
        path: '/app/example/page-3',
      },
    ],
  },
];
const location = object('location', {
  pathname: '/app/example/page-1',
}) as Location;
const pathname = '/app/example';
const defaultOpenKeys = [pathname];
const flatMenuKeys = getFlatMenuKeys(menu);

storiesOf('SideBar', module)
  .addDecorator((story) => <BrowserRouter>{story()}</BrowserRouter>)
  .add('default', () => (
    <SideBar width={320} collapsed={collapsed}>
      <SideBarHeader name="Aurora" logo={logo} />
      <BaseMenu
        menu={menu}
        openKeys={[]}
        inlineIndent={12}
        location={location}
        flatMenuKeys={flatMenuKeys}
        style={{ marginTop: 16 }}
      />
      <ChannelMenu title="Team" list={channels} onRemove={action('onRemove')} />
      <BaseMenu
        menu={jobsMenu}
        location={location}
        openKeys={defaultOpenKeys}
        flatMenuKeys={flatMenuKeys}
        defaultOpenKeys={defaultOpenKeys}
        onOpenChange={action('onPenChange')}
      />
    </SideBar>
  ))
  .add('Mobile', () => (
    <SideBar width={320} collapsed={collapsed}>
      <SideBarHeader name="Aurora" logo={logo} isMobile={true} />
      <BaseMenu
        menu={menu}
        openKeys={[]}
        inlineIndent={12}
        location={location}
        style={{ marginTop: 16 }}
        flatMenuKeys={flatMenuKeys}
      />
      <ChannelMenu title="Team" list={channels} onRemove={action('onRemove')} />
      <BaseMenu
        menu={jobsMenu}
        location={location}
        openKeys={defaultOpenKeys}
        flatMenuKeys={flatMenuKeys}
        defaultOpenKeys={defaultOpenKeys}
        onOpenChange={action('onPenChange')}
      />
    </SideBar>
  ))
  .add('SidBar in Candidate mobile', () => (
    <SideBar width={320} collapsed={collapsed}>
      <SideBarHeader name="Aurora" logo={logo} isMobile={true} productId={ProductType.Candidate} />
      <ChannelMenu title="Team" list={channels} onRemove={action('onRemove')} />
      <BaseMenu
        menu={jobsMenu}
        location={location}
        openKeys={defaultOpenKeys}
        flatMenuKeys={flatMenuKeys}
        defaultOpenKeys={defaultOpenKeys}
        onOpenChange={action('onPenChange')}
      />
    </SideBar>
  ))
  .add('SidBar in Candidate', () => (
    <SideBar width={320} collapsed={collapsed}>
      <SideBarHeader name="Aurora" logo={logo} productId={ProductType.Candidate} />
      <ChannelMenu title="Team" list={channels} onRemove={action('onRemove')} />
      <BaseMenu
        menu={jobsMenu}
        location={location}
        openKeys={defaultOpenKeys}
        flatMenuKeys={flatMenuKeys}
        defaultOpenKeys={defaultOpenKeys}
        onOpenChange={action('onPenChange')}
      />
    </SideBar>
  ))
  .add('CompanyList in Candidate', () => <CompanyModal visible list={companyList} companyId="123" />);
