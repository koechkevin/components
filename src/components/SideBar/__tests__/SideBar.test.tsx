import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import SideBar from '../SideBar';

describe('SideBar Component Test', () => {
  it('renders with child element', () => {
    const tree = shallow(<SideBar>SideBar Test</SideBar>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders side bar with prop width', () => {
    const tree = shallow(<SideBar width={200}>Button Test</SideBar>);
    const props = tree.props();

    expect(props.width).toEqual(200);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders dark theme with theme value "dark"', () => {
    const tree = shallow(<SideBar theme="dark" />);
    const props = tree.props();

    expect(props.theme).toEqual('dark');
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders light theme with theme value "light"', () => {
    const tree = shallow(<SideBar theme="light" />);
    const props = tree.props();

    expect(props.theme).toEqual('light');
    expect(toJson(tree)).toMatchSnapshot();
  });
});
