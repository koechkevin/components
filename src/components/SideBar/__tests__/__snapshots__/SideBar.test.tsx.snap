// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component Test renders dark theme with theme value "dark" 1`] = `
<Sider
  className="sideBar "
  collapsedWidth={0}
  collapsible={true}
  theme="dark"
  trigger={null}
/>
`;

exports[`SideBar Component Test renders light theme with theme value "light" 1`] = `
<Sider
  className="sideBar "
  collapsedWidth={0}
  collapsible={true}
  theme="light"
  trigger={null}
/>
`;

exports[`SideBar Component Test renders side bar with prop width 1`] = `
<Sider
  className="sideBar "
  collapsedWidth={0}
  collapsible={true}
  trigger={null}
  width={200}
>
  Button Test
</Sider>
`;

exports[`SideBar Component Test renders with child element 1`] = `
<Sider
  className="sideBar "
  collapsedWidth={0}
  collapsible={true}
  trigger={null}
>
  SideBar Test
</Sider>
`;
